web/core/src/jsMain/kotlin/org/jetbrains/compose/web/events/SyntheticAnimationEvent.kt

```diff
@@ -0,0 +1,14 @@
+package org.jetbrains.compose.web.events
+
+import androidx.compose.web.events.SyntheticEvent
+import org.w3c.dom.events.Event
+import org.w3c.dom.events.EventTarget
+
+class SyntheticAnimationEvent(
+    nativeEvent: Event,
+) : SyntheticEvent<EventTarget>(nativeEvent) {
+
+    val animationName: String = nativeEvent.asDynamic().animationName.unsafeCast<String>()
```

Comment:
let's have external interface instead of asDynamic casts

Reply:
```kotlin
external interface AnimationEventDetails {
    val animationName: String
}
```

`nativeEvent` is of type `Event`, which is an external class, so our external interface can't inherit from the Event class.
So I added one more parameter to the constructor:

```kotlrin
constructor(nativeEvent: Event, animationEventDetails: AnimationEventDetails)
```

and I use it like this:

```kotlin
SyntheticAnimationEvent(nativeEvent, nativeEvent.unsafeCast<AnimationEventDetails>())
```